### PR TITLE
Implement warning when ADL 1.4 incompatible codes are used

### PR DESCRIPTION
**Pull request by pieterbos**
To be able to convert back to ADL 1.4 node ids/value codes/value set codes, the numbers of the archetypes must not overlap. So, if id5 is used, at5 and ac5 should not be used. The generators in Archie did not generate this already, so it should be minor.

This PR adds a warning to the archetype validator in case overlap in numbers happens.

### CODE REVIEW

**Comment by pieterbos** [comment]
Not mergeable yet - the error that should be triggered is not covered, according to codecov, meaning the testing mechanism does not seem to work.

**Comment by pieterbos** [comment]
Although the problem is smaller, this also needs to check ac-codes against id and at codes

File: tools/src/test/resources/adl2-tests/validity/basics/openEHR-EHR-OBSERVATION.WARN_adl14_incompatible_node_ids.v1.0.0.adls
Comment (J3173) [inline tools/src/test/resources/adl2-tests/validity/basics/openEHR-EHR-OBSERVATION.WARN_adl14_incompatible_node_ids.v1.0.0.adls:1]: Also this file has mixed tabs and spaces.

**Review by J3173 — APPROVED** [review]
Nice!

**Pull request merged**